include/aspect/particle/property/melt_particle.h

```diff
@@ -0,0 +1,122 @@
+/*
+ Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_particle_property_melt_particle_h
+#define _aspect_particle_property_melt_particle_h
+
+#include <aspect/particle/property/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * Implementation of a plugin in which the particle
+       * property is defined as presence of melt above the
+       * melt transport threshold. Returns a porosity of 0
+       * if melt is not present and a porosity of 1 if melt
+       * is present.
+       *
+       * @ingroup ParticleProperties
+       */
+      template <int dim>
+      class Melt : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
```

Comment:
Usually we try to align the file names with the class names that are used inside of them. This would mean either renaming the files into melt.cc and melt.h, which I do not like, because we have files with the same names in other folder already. Or renaming the class and all its functions to `MeltParticle`, which I like more, because it better describes what this property is meant for. It is not meant to track the porosity value, it is meant to distinguish between `MeltParticle`'s and normal particles.